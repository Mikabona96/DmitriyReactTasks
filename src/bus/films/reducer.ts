// types
import { types } from './types';

const initialState = {
    data:        null,
    isFetching:  false,
    error:       null,
    currentFilm: null,
};

export const filmReducer = (state: any = initialState, { type, payload }: any): any => {
    switch (type) {
        case types.FILM_START_FETCHING:
            return {
                ...state,
                isFetching: true,
            };
        case types.FILM_STOP_FETCHING:
            return {
                ...state,
                isFetching: false,
            };
        case types.FILM_SET_FETCHING_ERROR:
            return {
                ...state,
                error: payload,
                data:  null,
            };
        case types.FILM_FILL:
            return {
                ...state,
                data:  payload,
                error: null,
            };

        case types.FILM_SET_CURRENT_FILM:
            return {
                ...state,
                currentFilm: payload,
            };


        default:
            return state;
    }
};
