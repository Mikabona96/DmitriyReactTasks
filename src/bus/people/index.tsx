/* eslint-disable react/no-unused-prop-types */
import React, { FC } from 'react';
import { usePeopleFetch } from './hooks/usePeopleFetch';
import { Outlet, useNavigate } from 'react-router-dom';
import { PeopleSection, PeoplePopup, ListJSX } from './styles';
import { useDispatch } from 'react-redux';
import { peopleActions } from './actions';

export const People: FC = (): any=>{
    const peopleFetch = usePeopleFetch();
    const navigate = useNavigate();
    const dispatch = useDispatch();
    if (peopleFetch) {
        const { isFetching, data, error } = peopleFetch;
        if (error && error.status === 404) {
            return <p>Not Found !</p>;
        }

        if (error && error.status !== 404) {
            return <p>Something went wrong</p>;
        }

        const spinnerJSX = isFetching && (
            <p>Loading data from api</p>
        );

        const listJSX = !isFetching && data && data.length && data.map((person: any, index: number)=>{
            const idRegExp = /\/([0-9]*)\/$/;
            const id =  person.url.match(idRegExp)[ 1 ];

            return (
                <ListJSX
                    key = { String(index) }
                    onClick = { () => {
                        dispatch(peopleActions.setCurrentPerson(person));
                        navigate(`/lesson7/people/${id}`);
                    } }>{person.name}
                </ListJSX>
            );
        });

        return (
            <PeopleSection>
                <div>
                    {spinnerJSX}
                    {listJSX}
                </div>
                <PeoplePopup>
                    <Outlet />
                </PeoplePopup>
            </PeopleSection>
        );
    }

    return <div>Ooops</div>;
};
