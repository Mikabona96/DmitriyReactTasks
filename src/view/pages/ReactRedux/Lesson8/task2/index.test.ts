import { peopleReducer } from '../../../../../bus/people/reducer';

const initialState = {
    data:          null,
    isFetching:    false,
    error:         null,
    currentPerson: null,
};

const action = {
    type:    'PEOPLE_START_FETCHING',
    payload: 'fdd',
};


test('sdsdsddsdsdsd', () => {
    expect(peopleReducer(initialState, action)).toMatchSnapshot();
});
