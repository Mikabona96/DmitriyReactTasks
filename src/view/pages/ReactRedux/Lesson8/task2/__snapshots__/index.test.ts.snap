// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sdsdsddsdsdsd 1`] = `
Object {
  "currentPerson": null,
  "data": null,
  "error": null,
  "isFetching": true,
}
`;
